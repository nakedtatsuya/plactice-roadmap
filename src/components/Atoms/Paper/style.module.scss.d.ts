declare const styles: {
  readonly root: string;
  readonly elevation0: string;
  readonly elevation1: string;
  readonly elevation2: string;
  readonly elevation3: string;
  readonly elevation4: string;
  readonly elevation5: string;
  readonly elevation6: string;
  readonly elevation7: string;
  readonly elevation8: string;
  readonly elevation9: string;
  readonly elevation10: string;
  readonly outlined: string;
  readonly rounded: string;
};
export = styles;
