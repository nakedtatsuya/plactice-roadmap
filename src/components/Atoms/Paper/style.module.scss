@import "../../properties.css";

.root {
  color: rgba(0, 0, 0, 0.87);
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  background-color: #fff;
}

.elevation0 {
  box-shadow: none;
}

.elevation1 {
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2), 0px 1px 1px 0px rgba(0,0,0,0.14), 0px 1px 3px 0px rgba(0,0,0,0.12);

}

.elevation2 {
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2), 0px 2px 2px 0px rgba(0,0,0,0.14), 0px 1px 5px 0px rgba(0,0,0,0.12);
}

.elevation3 {
  box-shadow: 0px 3px 3px -2px rgba(0,0,0,0.2), 0px 3px 4px 0px rgba(0,0,0,0.14), 0px 1px 8px 0px rgba(0,0,0,0.12);
}

.elevation4 {
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2), 0px 4px 5px 0px rgba(0,0,0,0.14), 0px 1px 10px 0px rgba(0,0,0,0.12);
}

.elevation5 {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2), 0px 5px 8px 0px rgba(0,0,0,0.14), 0px 1px 14px 0px rgba(0,0,0,0.12);
}

.elevation6 {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2), 0px 6px 10px 0px rgba(0,0,0,0.14), 0px 1px 18px 0px rgba(0,0,0,0.12);
}

.elevation7 {
  box-shadow: 0px 4px 5px -2px rgba(0,0,0,0.2), 0px 7px 10px 1px rgba(0,0,0,0.14), 0px 2px 16px 1px rgba(0,0,0,0.12);
}

.elevation8 {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2), 0px 8px 10px 1px rgba(0,0,0,0.14), 0px 3px 14px 2px rgba(0,0,0,0.12);
}

.elevation9 {
  box-shadow: 0px 5px 6px -3px rgba(0,0,0,0.2), 0px 9px 12px 1px rgba(0,0,0,0.14), 0px 3px 16px 2px rgba(0,0,0,0.12);
}

.elevation10 {
  box-shadow: 0px 6px 6px -3px rgba(0,0,0,0.2), 0px 10px 14px 1px rgba(0,0,0,0.14), 0px 4px 18px 3px rgba(0,0,0,0.12);
}

.outlined {
  border: var(--border);
}

.rounded {
  border-radius: var(--radius);
}
