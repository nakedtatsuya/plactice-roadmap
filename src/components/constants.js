export const SM_QUERY = 'screen and (max-width: 544px)'
export const MD_QUERY = 'screen and (max-width: 767px)'
export const LG_QUERY = 'screen and (max-width: 991px)'
export const XL_QUERY = 'screen and (max-width: 1199px)'
