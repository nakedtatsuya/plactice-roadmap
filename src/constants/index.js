export const MEDIA_QUERY_SM = 'screen and (max-width: 544px)'
export const MEDIA_QUERY_MD = 'screen and (max-width: 767px)'
export const MEDIA_QUERY_LG = 'screen and (max-width: 991px)'
export const MEDIA_QUERY_XL = 'screen and (max-width: 1199px)'
