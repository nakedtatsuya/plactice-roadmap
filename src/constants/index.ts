export enum TxtRole {
  base = 'base',
  white = 'white',
  info = 'info',
  success = 'success',
  warning = 'warning',
  danger = 'danger',
  primary = 'primary',
}

export enum ElevationLevel {
  zero = 'elevation0',
  one = 'elevation1',
  two = 'elevation2',
  three = 'elevation3',
  fore = 'elevation4',
  five = 'elevation5',
  six = 'elevation6',
  seven = 'elevation7',
  eight = 'elevation8',
  nine = 'elevation9',
  ten = 'elevation10',
}

export enum ButtonType {
  base = 'base',
  primary = 'primary',
  secondary = 'secondary',
  warning = 'warning',
  danger = 'danger',
}

export enum FontSize {
  xxs = 'xxs',
  xs = 'xs',
  s = 's',
  m = 'm',
  l = 'l',
  xl = 'xl',
  xxl = 'xxl',
  xxxl = 'xxxl',
  xxxxl = 'xxxxl',
}

export enum TagType {
  p = 'p',
  span = 'span',
  div = 'div',
}

export enum Level {
  h1 = 'h1',
  h2 = 'h2',
  h3 = 'h3',
  h4 = 'h4',
  h5 = 'h5',
  h6 = 'h6',
}

export enum ImageName {
  rocketImage = 'rocketImage',
  successImage = 'successImage',
  milestoneImage = 'milestoneImage',
  goalImage = 'goalImage',
  selectMilestoneImage = 'selectMilestoneImage',
  aroundImage = 'aroundImage',
  deepLearnImage = 'deepLearnImage',
  outputImage = 'outputImage',
  whyImage = 'whyImage',
  pathNodeImage = 'pathNodeImage',
}

export enum Milestones {
  milestoneImage = 'milestoneImage',
}

export enum AboutImage {
  goalImage = 'goalImage',
  selectMilestoneImage = 'selectMilestoneImage',
  aroundImage = 'aroundImage',
  deepLearnImage = 'deepLearnImage',
  outputImage = 'outputImage',
  whyImage = 'whyImage',
  pathNodeImage = 'pathNodeImage',
}

export enum IconName {
  goal = 'goal',
  light = 'light',
  stairs = 'stairs',
  test = 'test',
  left = 'left',
  rightTriangle = 'rightTriangle',
}

export enum LogoName {
  headerLogo = 'headerLogo',
  inlineLogo = 'inlineLogo',
  whiteLogo = 'whiteLogo',
}
