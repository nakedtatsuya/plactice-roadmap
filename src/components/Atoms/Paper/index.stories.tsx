import {storiesOf} from "@storybook/react";
import Paper from "./index";
import React from "react";
import {ElevationLevel} from "../../../constants";

storiesOf(`Atoms/Paper`, module)
.add(`default`, () => (
		<Paper style={{height: '128px', width: '128px', margin: '8px'}}/>
))
.add(`elevation0`, () => (
		<Paper style={{height: '128px', width: '128px', margin: '8px'}}/>
))
.add(`elevation1`, () => (
		<Paper elevation={ElevationLevel.one} style={{height: '128px', width: '128px', margin: '8px'}}/>
))
.add(`elevation2`, () => (
		<Paper elevation={ElevationLevel.two} style={{height: '128px', width: '128px', margin: '8px'}}/>
))
.add(`elevation3`, () => (
		<Paper elevation={ElevationLevel.three} style={{height: '128px', width: '128px', margin: '8px'}}/>
))
.add(`elevation4`, () => (
		<Paper elevation={ElevationLevel.fore} style={{height: '128px', width: '128px', margin: '8px'}}/>
))
.add(`elevation5`, () => (
		<Paper elevation={ElevationLevel.five} style={{height: '128px', width: '128px', margin: '8px'}}/>
))
.add(`elevation6`, () => (
		<Paper elevation={ElevationLevel.six} style={{height: '128px', width: '128px', margin: '8px'}}/>
))
.add(`elevation7`, () => (
		<Paper elevation={ElevationLevel.seven} style={{height: '128px', width: '128px', margin: '8px'}}/>
))
.add(`elevation8`, () => (
		<Paper elevation={ElevationLevel.eight} style={{height: '128px', width: '128px', margin: '8px'}}/>
))
.add(`elevation9`, () => (
		<Paper elevation={ElevationLevel.nine} style={{height: '128px', width: '128px', margin: '8px'}}/>
))
.add(`elevation10`, () => (
		<Paper elevation={ElevationLevel.ten} style={{height: '128px', width: '128px', margin: '8px'}}/>
))
.add(`square`, () => (
		<Paper square style={{height: '128px', width: '128px', margin: '8px'}}/>
))
.add(`outlined`, () => (
		<Paper variant={'outlined'} elevation={ElevationLevel.ten} style={{height: '128px', width: '128px', margin: '8px'}}/>
))
