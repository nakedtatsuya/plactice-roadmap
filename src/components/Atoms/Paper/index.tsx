import React, {ElementType} from "react";
import {containPresenter} from "../../utils/HoC";
import style from './style.module.scss';
import {ElevationLevel} from "../../../constants";

interface IProps extends React.HTMLAttributes<HTMLElement>{
		elevation?: ElevationLevel;
		variant?: 'elevation' | 'outlined';
		square?: boolean;
		tag?: ElementType;
}

const PaperPresenter: React.FC<IProps> = ({tag: Tag, className, ...props}) => (
		<Tag
				className={className}
				{...props}
		/>
);



interface IPropsContainer extends IProps {
		presenter: (presenterProps: IProps) => JSX.Element
}

export const PaperContainer: React.FC<IPropsContainer> = ({presenter, elevation, variant, square, className, ...props}) => {

		if(variant === 'outlined') {
				className = [className, style.outlined].join(' ');
		} else {
				className = [className, style[elevation]].join(' ');
		}

		if(!square) className += ` ${style.rounded}`;

		return presenter({className, ...props})
};



const Paper: React.FC<IProps> = containPresenter(PaperContainer, PaperPresenter);
Paper.defaultProps = {
		elevation: ElevationLevel.one,
		tag: 'div',
		variant: 'elevation',
		square: false,
};

export default Paper;
