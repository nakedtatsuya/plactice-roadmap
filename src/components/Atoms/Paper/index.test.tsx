import React from 'react';
import renderer from 'react-test-renderer';
import Paper from './index';
import { ElevationLevel } from '../../../constants';

describe('Paper', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Paper className={'className'}>ペーパー</Paper>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Paper logic test', () => {
  it('default className', () => {
    const result = renderer
      .create(<Paper>button</Paper>)
      .toJSON()
      .props.className.split(' ');
    expect(['', 'elevation1', 'rounded']).toEqual(result);
  });

  it('custom elevation className ', () => {
    const result = renderer
      .create(<Paper elevation={ElevationLevel.seven}>button</Paper>)
      .toJSON()
      .props.className.split(' ');
    expect(['', 'elevation7', 'rounded']).toEqual(result);
  });

  it('custom elevation className and outlined', () => {
    const result = renderer
      .create(
        <Paper variant={'outlined'} elevation={ElevationLevel.seven}>
          button
        </Paper>
      )
      .toJSON()
      .props.className.split(' ');
    expect(['', 'outlined', 'rounded']).toEqual(result);
  });

  it('not rounded class', () => {
    const result = renderer
      .create(<Paper square>button</Paper>)
      .toJSON()
      .props.className.split(' ');
    expect(['', 'elevation1']).toEqual(result);
  });
});
