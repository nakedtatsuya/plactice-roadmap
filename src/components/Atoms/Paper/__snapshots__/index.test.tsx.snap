// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paper renders correctly 1`] = `
<div
  className="className elevation1 rounded"
>
  ペーパー
</div>
`;
